packages/cli/src/utils/checkpoint.ts

```diff
@@ -0,0 +1,178 @@
+import Debug from '@prisma/debug'
+import { getCLIPathHash, getConfig, getProjectHash, getSchema, parseEnvValue } from '@prisma/sdk'
+import type { Check } from 'checkpoint-client'
+import * as checkpoint from 'checkpoint-client'
+
+const debug = Debug('prisma:cli:checkpoint')
+
+/**
+ * Collect and prepare data then run the Checkpoint Client which will send some info to the remote Checkpoint Server
+ * It will be done in a child process so the CLI won't have to wait for it to finish to exit
+ *
+ * returns if the current CLI version is outdated.
+ *
+ * For more info about the data collected by the Checkpoint Server and how to disable it see:
+ * https://www.prisma.io/docs/concepts/more/telemetry
+ */
+export async function runCheckpointClientCheck({
+  schemaPath,
+  isPrismaInstalledGlobally,
+  version,
+  commandArray,
+  telemetryInformation,
+}: {
+  schemaPath: string
+  isPrismaInstalledGlobally: 'npm' | 'yarn' | false
```

Comment:
ugh

Reply:
That's the type, a very weird boolean

Reply:
Oh I know. It is just really terrible and bad, but of course works and I know how it got this way.